src/Xamarin.Android.Build.Tasks/Xamarin.Android.Build.Tasks.targets

```diff
@@ -290,4 +290,45 @@
     />
     <Exec Command="chmod +x @(_MonoScriptDestination->'%(Identity)', ' ')" />
   </Target>
+
+  <ItemGroup>
+    <InputAssemblies Include="$(OutputPath)\Newtonsoft.Json.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Common.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Configuration.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.DependencyResolver.Core.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Frameworks.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.LibraryModel.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Packaging.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Packaging.Core.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.ProjectModel.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Protocol.dll" />
+    <InputAssemblies Include="$(OutputPath)\NuGet.Versioning.dll" />
+    <InputAssemblies Include="$(OutputPath)\System.Collections.Immutable.dll" />
+    <InputAssemblies Include="$(OutputPath)\System.Reflection.Metadata.dll" />
+  </ItemGroup>
+
+  <Target Name="ILRepacker"
+      AfterTargets="CopyFilesToOutputDirectory"
+      Inputs="$(OutputPath)\$(AssemblyName).dll;@(InputAssemblies)"
+      Outputs="$(OutputPath)\$(AssemblyName).dll;$(IntermediateOutputPath)ILRepacker.stamp"
```

Comment:
Putting the same `$(OutputPath)\$(AssemblyName).dll` file in `Inputs` and `Outputs` is probably not needed. You should be able to just rely on the stamp file.

Reply:
Updated